pkg/agent/agent_readiness.go

```diff
@@ -26,9 +27,9 @@ type PodState struct {
 
 // AllReachedGoalState returns whether the agents associated with a given StatefulSet have reached goal state.
 // it achieves this by reading the Pod annotations and checking to see if they have reached the expected config versions.
-func AllReachedGoalState(sts appsv1.StatefulSet, podGetter pod.Getter, desiredMemberCount, targetConfigVersion int, log *zap.SugaredLogger) (bool, error) {
+func AllReachedGoalState(ctx context.Context, sts appsv1.StatefulSet, podGetter pod.Getter, desiredMemberCount, targetConfigVersion int, log *zap.SugaredLogger) (bool, error) {
```

Comment:
I appreciate this but this is something we probably need a discuss in our team. We figure out a way to introduce API breaking changes without too much of a problem. Otherwise, it will be hard to evolve the operator.

Reply:
That's the reason I like monorepo approach better than multiple github repos :) Definitely agreed. Anyway, such change should be probably accompanied by PR to enterprise with bump to community version.